app/logic/validator_score_v2_logic.rb

```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+
+# Logic to compile ValidatorScoreV2
+module ValidatorScoreV2Logic
+  include PipelineLogic
+
+  # The constant below defines into how many groups we divide validators for scoring. The first group
+  # will be assigned with score 0, the next one with +1, and so on, until we reach the limit of groups.
+  NUMBER_OF_GROUPS = 3
+
+  # Payload starts with :network & :batch_uuid
+  def set_this_batch
```

Comment:
I'll leave it for now, because it's the method name we use in v1. It'll be easier to compare and tell which code we need t copy or not, if we decide we want to fully switch to v2 and drop v1.